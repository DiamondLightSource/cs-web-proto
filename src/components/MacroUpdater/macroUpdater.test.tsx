import React from "react";
import { MacroUpdater } from "./macroUpdater";
import { configure, mount, ReactWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { create, ReactTestRenderer } from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();

let snapshot: ReactTestRenderer;
let wrapper: ReactWrapper;

beforeEach((): void => {
  const store = mockStore({
    macroMap: {
      a: "b"
    }
  });
  wrapper = mount(
    <Provider store={store}>
      <MacroUpdater />
    </Provider>
  );
  snapshot = create(
    <Provider store={store}>
      <MacroUpdater />
    </Provider>
  );
});

describe("<Input />", (): void => {
  test("it matches the snapshot", (): void => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
  test("it renders two inputs", (): void => {
    expect(wrapper.find("input").length).toEqual(2);
  });
});
