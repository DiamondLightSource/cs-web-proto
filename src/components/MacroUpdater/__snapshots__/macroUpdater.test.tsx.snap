// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> it matches the snapshot 1`] = `
Array [
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />,
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />,
]
`;
